### PR TITLE
Add Pub/Sub compression handling to ingestion-sink

### PR DESCRIPTION
**Pull request by relud**

### CODE REVIEW

File: ingestion-sink/src/main/java/com/mozilla/telemetry/ingestion/sink/config/SinkConfig.java [hunk 1]
```diff
@@ -117,10 +121,16 @@ public Output via(Function<PubsubMessage, CompletableFuture<Void>> write) {
   enum OutputType {
     pubsub {
 
+      private CompressPayload getOutputCompression(Env env) {
+        return CompressPayload.valueOf(env.getString(OUTPUT_COMPRESSION, "none"));
```
Comment (jklukas) [inline ingestion-sink/src/main/java/com/mozilla/telemetry/ingestion/sink/config/SinkConfig.java]:
```suggestion
        return CompressPayload.valueOf(env.getString(OUTPUT_COMPRESSION, CompressPayload.none.toString()));
```

File: ingestion-sink/src/main/java/com/mozilla/telemetry/ingestion/sink/transform/CompressPayload.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import com.google.protobuf.ByteString;
+import com.google.pubsub.v1.PubsubMessage;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.function.Function;
+import java.util.zip.GZIPOutputStream;
+
+public enum CompressPayload implements Function<PubsubMessage, PubsubMessage> {
+  gzip {
```
Comment (jklukas) [inline ingestion-sink/src/main/java/com/mozilla/telemetry/ingestion/sink/transform/CompressPayload.java]: Do we have other enums in this codebase yet? I'm feeling some regret about the decision to use lowercase enum values in ingestion-beam. It looks nicer on the command-line, but breaks with the style used by enum options used by Beam and breaks with established Java convention for all-caps enum values.

Comment (relud) [reply ingestion-sink/src/main/java/com/mozilla/telemetry/ingestion/sink/transform/CompressPayload.java]: we have `PubsubMessageToObjectNode.Format`, which I'm planning to refactor as part of implementing `OUTPUT_FORMAT=payload`. I'm fine with switching to all-caps.

Comment (relud) [reply ingestion-sink/src/main/java/com/mozilla/telemetry/ingestion/sink/transform/CompressPayload.java]: I switched both enums to use all-caps values

Comment (relud) [reply ingestion-sink/src/main/java/com/mozilla/telemetry/ingestion/sink/transform/CompressPayload.java]: (and use `.toUpperCase()`, for backward compatiblity)

**Pull request merged**